torchrl/envs/libs/gym.py

```diff
@@ -172,10 +173,26 @@ def lib(self) -> ModuleType:
         return gym
 
     def _set_seed(self, seed: int) -> int:
-        if version.parse(gym.__version__) < version.parse("0.19.0"):
-            self._env.seed(seed=seed)
-        else:
+        skip = False
+        if self._seed_calls_reset is None:
+            if version.parse(gym.__version__) < version.parse("0.19.0"):
+                self._seed_calls_reset = False
+                self._env.seed(seed=seed)
+            else:
+                try:
+                    self.reset(seed=seed)
+                    skip = True
+                    self._seed_calls_reset = True
+                except TypeError as err:
+                    warnings.warn(
+                        f"reset with seed kwarg returned an exception: {err}.\n"
+                        f"Calling env.seed from now on."
+                    )
+                    self._seed_calls_reset = False
+        if self._seed_calls_reset and not skip:
             self.reset(seed=seed)
+        else:
+            self._env.seed(seed=seed)
```

Comment:
If `self._seed_calls_reset`  is `False`, wouldn't this line still error out as the `_env` does not accept a seed? i.e., `TypeError: reset() got an unexpected keyword argument 'seed'`